synthesizer/program/src/resources/credits.aleo

```diff
@@ -106,573 +149,515 @@ record credits:
     // Output the finalize future.
-    output r7 as credits.aleo/bond_public.future;
+    output r5 as credits.aleo/bond_validator.future;
 
-finalize bond_public:
-    // Input the staker's address.
-    input r0 as address.public;
-    // Input the validator's address.
-    input r1 as address.public;
-    // Input the withdrawal address.
-    input r2 as address.public;
-    // Input the amount of microcredits to bond.
-    input r3 as u64.public;
+finalize bond_validator:
+    input r0 as address.public;  // Validator's address.
+    input r1 as address.public;  // Withdrawal address.
+    input r2 as u64.public;      // Amount of microcredits to bond.
+    input r3 as u8.public;       // Commission percent.
 
     // Retrieve the withdrawal address for the staker.
-    get.or_use withdraw[r0] r2 into r4;
+    get.or_use withdraw[r0] r1 into r4;
     // Ensure that the withdrawal address is consistent.
-    assert.eq r2 r4;
-
-    // Determine whether the caller is a validator.
-    is.eq r0 r1 into r5;
-    // If the caller is a validator, jump to the `bond_validator` logic.
-    branch.eq r5 true to bond_validator;
-    // If the caller is not a validator, jump to the `bond_delegator` logic.
-    branch.eq r5 false to bond_delegator;
-
-    /******* Bond Validator *******/
-
-    // Starts the `bond_validator` logic.
-    position bond_validator;
-
-    /* Committee */
-
-    // Check if the validator is already in the committee.
-    contains committee[r0] into r6;
-    // If the validator is already in the committee, jump to the `continue_bond_validator` logic.
-    branch.eq r6 true to continue_bond_validator;
-
-    // Get the committee size.
-    get.or_use metadata[aleo1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq3ljyzc] 0u32 into r7;
-    // Increment the committee size by one.
-    add r7 1u32 into r8;
-    // Set the new committee size.
-    set r8 into metadata[aleo1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq3ljyzc];
+    assert.eq r1 r4;
     // Set the withdrawal address.
-    set r2 into withdraw[r0];
-
-    // Continues the rest of the `bond_validator` logic.
-    position continue_bond_validator;
+    set r4 into withdraw[r0];
 
     // Construct the initial committee state.
     // Note: We set the initial 'is_open' state to 'true'.
-    cast 0u64 true into r9 as committee_state;
+    cast true r3 into r5 as committee_state;
     // Retrieve the committee state of the specified validator.
-    get.or_use committee[r0] r9 into r10;
+    get.or_use committee[r0] r5 into r6;
+
+    // Fail if attempting to change commission. Commission can only be set when validator bonds the first time.
+    assert.eq r3 r6.commission;
 
-    // Increment the stake for the specified validator.
-    add r10.microcredits r3 into r11;
     // Construct the updated committee state.
-    cast r11 r10.is_open into r12 as committee_state;
+    cast r6.is_open r6.commission into r7 as committee_state;
```

Comment:
This operation is technically unnecessary. You can use `r6` (which is equivalent to `r7`).

Reply:
Resolved in [1a13a11](https://github.com/AleoNet/snarkVM/pull/2453/commits/1a13a11ebd17b828ca7b8fbcbe70b3e38b8505e0)